### ISSUE TITLE
Plugin 'addDataExport' - incorrect export of parent/ child hierarchy 

### ISSUE THREAD

**Issue by BlueFoxPrime** [bryanmylee/svelte-headless-table#61]
Hi :)

I noticed that the plugin does not seem to output parent/ child hierarchy correctly when choosing type 'object':
![image](https://user-images.githubusercontent.com/112826784/193262299-33e94464-272d-410c-ae2a-820695a3426b.png)

Expected behaviour in this case would be the that there are 2 top level objects instead of 6, when expanded.
[Example RPL
](https://svelte.dev/repl/418ea92aeb814dbd8056bd508978f9d2?version=3.50.1)

In case the error is on my end.. please give me a hint.
Your awesome work is very much appreciated!

**Comment by bryanmylee** [comment]
Hey @BlueFoxPrime, unfortunately this was a decision that I decided to make for the data export plugin.

My reasoning was that the export plugin is primarily concerned with exporting data as it looks in the table.

However, if there's a need for it, I can look into extending the plugin with an option to preserve nesting?

**Comment by bryanmylee** [comment]
@BlueFoxPrime Let me know if the nested behavior is what you need, and if the use case is common enough I'll work on adding an option to enable such a feature!

**Comment by BlueFoxPrime** [comment]
Hey, sorry for the late response but I am on holiday :)

I think that the use case is quite common because I want to share data to other component(s) depending on what’s displayed in the table. The way it’s designed produces strange results, as children are also displayed as parent items. I did some minor modifications to the ‘addDataExport’ plugin to change it. I didn’t manage to test it yet but it seems to work just fine. I can post the code when I get back.

**Comment by bryanmylee** [comment]
Hey @BlueFoxPrime, that sounds great!

Most of the plugins are missing a lot of test coverage at the moment. If you open a PR, I'll gladly write the testcases for it.

**Comment by BlueFoxPrime** [comment]
Hey, I am not sure I can put together a decent PR but I am trying and I will surely post my code here.
I figured that addFlatten and addPagination need to be respected. Therefore I wanted to use $pageRows instead of $rows to determine the derived store returned from the Plugin. Thereby I realized that the docs are possibly not correct / there is a bug regarding the following:

![grafik](https://user-images.githubusercontent.com/112826784/199714619-490c60c9-e0ec-44da-bb8f-73635abb87ac.png)

$pageRows is undefined if addPagination is not declared. 

Reproduction:
https://svelte.dev/repl/3be0d4e7915d4d5f9ab1f3b65e599a5d?version=3.52.0
